opensquirrel/passes/exporter/cqasmv1_exporter.py

```diff
@@ -62,9 +61,33 @@ def visit_gate(self, gate: Gate) -> None:
         if any(not isinstance(arg, Qubit) for arg in gate.arguments):  # type: ignore[union-attr]
             params = [arg.accept(self) for arg in gate.arguments if not isinstance(arg, Qubit)]  # type: ignore[union-attr]
         qubit_args = (arg.accept(self) for arg in gate.arguments if isinstance(arg, Qubit))  # type: ignore[union-attr]
-        self.cqasmv1_string += "{} {}{}\n".format(
-            gate_name, ", ".join(qubit_args), ", " + ", ".join(params) if params else ""
-        )
+        self.output += "{} {}{}\n".format(gate_name, ", ".join(qubit_args), ", " + ", ".join(params) if params else "")
+
+
+def post_process(output: str) -> str:
+    return _merge_barrier_groups(output)
+
+
+def _merge_barrier_groups(output: str) -> str:
+    post_output: str = ""
+    barrier_group_indices: list[int] = []
+    for line in output.split("\n"):
+        if not line.startswith("barrier"):
+            if barrier_group_indices:
+                post_output += _dump_barrier_group(barrier_group_indices)
+                barrier_group_indices = []
+            post_output += f"{line}\n"
+        else:
+            barrier_group_indices.append(_get_barrier_index(line))
+    return post_output
```

Comment:
`ret`, or `result` are generic return value variable names that can fit good in this occasion.

```suggestion
    ret: str = ""
    barrier_group_indices: list[int] = []
    for line in output.split("\n"):
        if not line.startswith("barrier"):
            if barrier_group_indices:
                ret += _dump_barrier_group(barrier_group_indices)
                barrier_group_indices = []
            ret += f"{line}\n"
        else:
            barrier_group_indices.append(_get_barrier_index(line))
    return ret
```